monai/transforms/spatial/array.py

```diff
@@ -425,16 +432,28 @@ def __call__(
             data_array (resampled into `self.pixdim`), original affine, current affine.
 
         """
-        sr = int(data_array.ndim - 1)
+        # if the input isn't MetaTensor and output isn't desired to be one either,
+        # nothing to do.
+        if not isinstance(data_array, MetaTensor) and not get_track_meta():
+            warnings.warn(
+                "`Orientation` applied to non-MetaTensor object and metadata tracking is off, transform does nothing."
+            )
+            return data_array
+
+        original_spatial_shape = data_array.shape[1:]
+        sr = len(original_spatial_shape)
         if sr <= 0:
             raise ValueError("data_array must have at least one spatial dimension.")
-        if affine is None:
+        affine_: np.ndarray
+        affine_np: np.ndarray
+        if isinstance(data_array, MetaTensor):
+            affine_np, *_ = convert_data_type(data_array.affine, np.ndarray)
+            affine_ = to_affine_nd(sr, affine_np)
+        else:
+            warnings.warn("`data_array` is not of type `MetaTensor, assuming affine to be identity.")
             # default to identity
-            affine_np = affine = np.eye(sr + 1, dtype=np.float64)
+            affine_np = np.eye(sr + 1, dtype=np.float64)
             affine_ = np.eye(sr + 1, dtype=np.float64)
```

Comment:
(might be better to reuse `MetaTensor.get_default_affine` in the future)

Reply:
Hm, default MetaTensor affine is always torch.eye(4), whereas here we allow for 3x3 etc matrices.